homeassistant/components/dwd_weather_warnings/coordinator.py

```diff
@@ -23,4 +24,11 @@ def __init__(self, hass: HomeAssistant, api: DwdWeatherWarningsAPI) -> None:
 
     async def _async_update_data(self) -> None:
         """Get the latest data from the DWD Weather Warnings API."""
-        await self.hass.async_add_executor_job(self.api.update)
+        if self.config_entry is not None:
```

Comment:
This will actually always be true. Type the class attribute as `ConfigEntry` instead.

Reply:
> Type the class attribute as ConfigEntry instead.

Sorry could you elaborate this part? I don't quite understand

Reply:
We can type class attributes like this, above the init method.
```py
config_entry: ConfigEntry

def __init__(self, ...) -> None:
```

Reply:
Am I understanding correctly that this makes `config_entry` **None** until it's set by the parent class, making the if condition I have valid?

Reply:
No, this will add the type annotation `ConfigEntry` to `config_entry` telling us it is always an instance of `ConfigEntry` and never `None`.

Reply:
Ah i see, I'll remove the check for `None` then, thanks

Reply:
Updated in 30d0fe3ed0de3df43b59ffdb03292116aa3dc46a